standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/utils/MetaStoreServerUtils.java

```diff
@@ -501,6 +502,28 @@ public static boolean areSameColumns(List<FieldSchema> oldCols, List<FieldSchema
     return ListUtils.isEqualList(oldCols, newCols);
   }
 
+  /**
+   * Returns true if p is a prefix of s.
+   */
+  public static boolean arePrefixColumns(List<FieldSchema> p, List<FieldSchema> s) {
+    if (p == s) {
+      return true;
+    }
+    if (p.size() > s.size()) {
+      return false;
+    }
+    Iterator itP = p.iterator();
```

Comment:
@kgyrtkirk Maybe we can use `ListUtils.isEqualList(p, p.subList(0, p.size()))`? that way we can avoid most of the code here?

Reply:
of course! :)